### PR TITLE
Feat/feedback

### PR DESCRIPTION
**Pull request by LittleNa1000**

### CODE REVIEW

File: app/api/feedback/route.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { addFeedback } from "@/lib/firebase/firestore";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function POST(req: NextRequest) {
+  try {
+    const data = await req.json();
+
+    if ((data.rating < 0 || data.rating > 5) && data.comment.trim() === "") {
+      throw new Error("Invalid input");
+    }
```
Comment (tontan2545) [inline app/api/feedback/route.ts:10]: This will throw 500 error to frontend, not a good practice kub. Try throwing 400 invalid request instead. 
```ts
return NextResponse.json(..., { status: 400 })
```

File: app/types/feedback.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export type TFeedback = {
```
Comment (tontan2545) [inline app/types/feedback.ts:1]: In typescript, it's not a commonly used convention to use `T` in front to label Type or `I` to label interface. That's usually for Java and C#. In this case the convention is to just do PascalCasing on the types. 

tl;dr: change from `TFeedback` to `Feedback`

File: components/ui/nextButton.tsx [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
-        <p className="fixed left-1/2 -translate-x-1/2 transform text-base font-semibold text-[#6C1F1F]">
-          ถัดไป
+      <Button
+        className="relative"
+        variant="next"
+        size="xs"
+        disabled={props.disabled}
+      >
+        <p className="fixed left-1/2 w-24 -translate-x-1/2 transform text-base font-semibold text-[#6C1F1F]">
+          {props.label ? props.label : "ถัดไป"}
```
Comment (tontan2545) [inline components/ui/nextButton.tsx:18]: consider using i18n for `ถัดไป`

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+                  strokeLinecap="round"
+                />
+                <path
+                  d="M14.6787 7.319C15.6962 8.29672 16.3397 8.43216 17.6623 8.02995"
+                  stroke={rating === 1 || rating === 0 ? "#C41C11" : "#C5C5C5"}
+                  strokeLinecap="round"
+                />
+                <path
+                  d="M7.36328 8.03918C8.75202 8.43604 9.38537 8.26327 10.3422 7.31081"
+                  stroke={rating === 1 || rating === 0 ? "#C41C11" : "#C5C5C5"}
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:120]: `rating < 2`

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    fetch(`${process.env.NEXT_PUBLIC_WEB_URL}/api/feedback`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        rating,
+        comment,
+      }),
+    });
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:39]: This fetch is not `await`-ed so the process finishes submission without knowing whether the request going alright or not.

tl;dr: because you did not await so next line runs right after the fetch request is sent.

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+    if (isSubmitting) return;
+    if (isEmpty) {
+      router.push("/end");
+      return;
+    }
+
+    setIsSubmitting(true);
+
+    fetch(`${process.env.NEXT_PUBLIC_WEB_URL}/api/feedback`, {
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:30]: You don't need to explicitly state the URL if you are sending request to it's own api route. In fact this will be broken in production.

Use: `fetch('/api/feedback')`

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+import "./styles.css";
+import SubmittedPage from "./submittedPage";
+
+type Props = {};
+
+const Page = (props: Props) => {
+  const [rating, setRating] = useState(0);
+  const [comment, setComment] = useState("");
+  const [submitted, setSubmitted] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:16]: can be shorten to

```ts
const [submissionStatus, setSubmissionStatus] = useState<'NOT_SUBMITTED' | 'SUBMITTING' | 'SUBMITTED'>('NOT_SUBMITTED')
```

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+    <>
+      <Image
+        src="/img/feedback.webp"
+        alt="Description"
+        layout="fill"
+        objectFit="cover"
+        className="fixed"
+      />
+      <form
+        action={process.env.NEXT_PUBLIC_WEB_URL + "/feedback"}
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:57]: This is server actions, try not to use this in combination with `onSubmit`

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+                width="25"
+                height="25"
+                viewBox="0 0 25 25"
+                fill="none"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  d="M12.5132 22.2121C18.1777 22.2121 22.7697 17.8706 22.7697 12.5151C22.7697 7.15966 18.1777 2.81818 12.5132 2.81818C6.84879 2.81818 2.25684 7.15966 2.25684 12.5151C2.25684 17.8706 6.84879 22.2121 12.5132 22.2121Z"
+                  stroke={rating === 1 || rating === 0 ? "#C41C11" : "#C5C5C5"}
+                  strokeWidth="2"
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:89]: Button 1 -> 5 looks like it can be done in a loop

File: app/[locale]/(non-interactive)/feedback/page.tsx [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+        comment,
+      }),
+    });
+
+    setSubmitted(true);
+    setIsSubmitting(false);
+  };
+  if (submitted) {
+    return <SubmittedPage />;
+  }
```
Comment (tontan2545) [inline app/[locale]/(non-interactive)/feedback/page.tsx:46]: why is this not a separate page?

**Review by tontan2545 — CHANGES_REQUESTED** [review]
Request change ngub

File: app/[locale]/(non-interactive)/feedback/page.tsx (hunk 9 above)
Comment (LittleNa1000) [reply app/[locale]/(non-interactive)/feedback/page.tsx:89]: พวก attribute value ใน svg ของแต่ละ btn มันไม่เหมือนกันอะครับ เลย hardcode ไปเลย

File: app/[locale]/(non-interactive)/feedback/page.tsx (hunk 10 above)
Comment (LittleNa1000) [reply app/[locale]/(non-interactive)/feedback/page.tsx:46]: อยากให้ submittedPage มันแสดงเฉพาะหลังกด submit แล้วเท่านั้นครับ

**Comment by tontan2545** [comment]
@LittleNa1000 fix merge commit then merge loey kub
